File: phi_intelligence/server/services/keyVaultService.ts
// Azure Key Vault removed - using environment variables for AWS deployment
// This service now acts as a simple environment variable accessor with caching

interface CachedSecret {
  value: string;
  expires: number;
}

class KeyVaultService {
  private cache: Map<string, CachedSecret> = new Map();
  private cacheTimeout = 5 * 60 * 1000; // 5 minutes

  constructor() {
    console.log('✅ KeyVault Service initialized (environment variables mode)');
  }

  async getSecret(secretName: string): Promise<string> {
    // Check cache first
    const cached = this.cache.get(secretName);
    if (cached && cached.expires > Date.now()) {
      return cached.value;
    }

    // Get from environment variable instead of Azure
    const envKey = secretName.toUpperCase().replace(/-/g, '_');
    const value = process.env[envKey] || '';
    
    if (!value) {
      console.warn(`⚠️ Environment variable ${envKey} not found for secret ${secretName}`);
      return '';
    }
    
    // Cache the value
    this.cache.set(secretName, {
      value,
      expires: Date.now() + this.cacheTimeout
    });
    
    return value;
  }

  // Database secrets
  async getDatabaseUrl(service: 'phi' | 'company'): Promise<string> {
    return this.getSecret(`database-${service}-url`);
  }

  // API keys
  async getOpenAIApiKey(): Promise<string> {
    return this.getSecret('openai-api-key');
  }

  async getDeepgramApiKey(): Promise<string> {
    return this.getSecret('deepgram-api-key');
  }

  async getPineconeApiKey(): Promise<string> {
    return this.getSecret('pinecone-api-key');
  }

  // LiveKit credentials
  async getLiveKitCredentials(project: string): Promise<{ apiKey: string; apiSecret: string; url: string }> {
    const [apiKey, apiSecret, url] = await Promise.all([
      this.getSecret(`livekit-${project}-api-key`),
      this.getSecret(`livekit-${project}-api-secret`),
      this.getSecret(`livekit-${project}-url`)
    ]);
    return { apiKey, apiSecret, url };
  }

  // R2 storage credentials
  async getR2Credentials(instance: 'primary' | 'secondary'): Promise<{
    accessKey: string;
    secretKey: string;
    bucketName: string;
    accountId: string;
  }> {
    const [accessKey, secretKey, bucketName, accountId] = await Promise.all([
      this.getSecret(`r2-${instance}-access-key`),
      this.getSecret(`r2-${instance}-secret-key`),
      this.getSecret(`r2-${instance}-bucket`),
      this.getSecret('cloudflare-account-id')
    ]);
    return { accessKey, secretKey, bucketName, accountId };
  }

  // JWT secrets
  async getJWTSecrets(): Promise<{ accessSecret: string; refreshSecret: string; sessionSecret: string }> {
    const [accessSecret, refreshSecret, sessionSecret] = await Promise.all([
      this.getSecret('jwt-access-secret'),
      this.getSecret('jwt-refresh-secret'),
      this.getSecret('session-secret')
    ]);
    return { accessSecret, refreshSecret, sessionSecret };
  }

  // Pinecone configuration
  async getPineconeConfig(): Promise<{ apiKey: string; environment: string; indexName: string }> {
    const [apiKey, environment, indexName] = await Promise.all([
      this.getSecret('pinecone-api-key'),
      this.getSecret('pinecone-environment'),
      this.getSecret('pinecone-index-name')
    ]);
    return { apiKey, environment, indexName };
  }

  // Clear cache
  clearCache(): void {
    this.cache.clear();
  }

  // Test connection (now just checks environment variables)
  async testConnection(): Promise<boolean> {
    try {
      const testKey = process.env.OPENAI_API_KEY;
      return !!testKey;
    } catch (error) {
      console.error('Environment variable access failed:', error);
      return false;
    }
  }
}

// Note: All methods now read from environment variables instead of Azure Key Vault

export default new KeyVaultService();

File: phi_voice/shared/key_vault_service.py
#!/usr/bin/env python3
"""
Azure Key Vault Service for PHI Voice Token Servers
"""

import os
import logging
import asyncio
from datetime import datetime, timedelta
from typing import Dict, Any, Optional, Tuple
from dataclasses import dataclass

try:
    from azure.identity import DefaultAzureCredential
    from azure.keyvault.secrets import SecretClient
    KEYVAULT_AVAILABLE = True
except ImportError:
    KEYVAULT_AVAILABLE = False
    logging.warning("⚠️ Azure Key Vault libraries not installed")

logger = logging.getLogger(__name__)

@dataclass
class CachedSecret:
    value: str
    expires: datetime

class KeyVaultService:
    def __init__(self):
        self.client = None
        self.cache: Dict[str, CachedSecret] = {}
        self.cache_timeout = timedelta(minutes=5)
        self.vault_url = "https://phi-intelligence-vault.vault.azure.net/"
        
        if KEYVAULT_AVAILABLE:
            try:
                credential = DefaultAzureCredential()
                self.client = SecretClient(vault_url=self.vault_url, credential=credential)
                logger.info("✅ Azure Key Vault service initialized")
            except Exception as e:
                logger.error(f"❌ Failed to initialize Key Vault service: {e}")
                self.client = None
        else:
            logger.warning("⚠️ Key Vault service not available")
    
    async def get_secret(self, secret_name: str) -> str:
        if not self.client:
            raise RuntimeError("Key Vault client not initialized")
        
        # Check cache first
        if secret_name in self.cache:
            cached = self.cache[secret_name]
            if datetime.now() < cached.expires:
                return cached.value
        
        try:
            secret = self.client.get_secret(secret_name)
            value = secret.value or ''
            
            # Cache the secret
            self.cache[secret_name] = CachedSecret(
                value=value,
                expires=datetime.now() + self.cache_timeout
            )
            
            return value
        except Exception as e:
            logger.error(f"Failed to get secret {secret_name}: {e}")
            raise
    
    # Database secrets
    async def get_database_url(self, service: str) -> str:
        return await self.get_secret(f"database-{service}-url")
    
    # API keys
    async def get_openai_key(self) -> str:
        return await self.get_secret("openai-api-key")
    
    async def get_deepgram_key(self) -> str:
        return await self.get_secret("deepgram-api-key")
    
    async def get_pinecone_key(self) -> str:
        return await self.get_secret("pinecone-api-key")
    
    # LiveKit credentials
    async def get_livekit_credentials(self, project: str) -> Tuple[str, str, str]:
        api_key = await self.get_secret(f"livekit-{project}-api-key")
        api_secret = await self.get_secret(f"livekit-{project}-api-secret")
        url = await self.get_secret(f"livekit-{project}-url")
        return api_key, api_secret, url
    
    # R2 storage credentials
    async def get_r2_credentials(self, instance: str) -> Tuple[str, str, str, str]:
        access_key = await self.get_secret(f"r2-{instance}-access-key")
        secret_key = await self.get_secret(f"r2-{instance}-secret-key")
        bucket_name = await self.get_secret(f"r2-{instance}-bucket")
        account_id = await self.get_secret("cloudflare-account-id")
        return access_key, secret_key, bucket_name, account_id
    
    # Pinecone configuration
    async def get_pinecone_config(self) -> Tuple[str, str, str]:
        api_key = await self.get_secret("pinecone-api-key")
        environment = await self.get_secret("pinecone-environment")
        index_name = await self.get_secret("pinecone-index-name")
        return api_key, environment, index_name
    
    # Clear cache
    def clear_cache(self):
        self.cache.clear()
    
    # Test connection
    async def test_connection(self) -> bool:
        try:
            await self.get_secret("openai-api-key")
            return True
        except Exception as e:
            logger.error(f"Key Vault connection test failed: {e}")
            return False

# Global instance
key_vault_service = KeyVaultService()
